### ISSUE TITLE
Nav Item

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#1054]
# Requirements 

- Orientation support (vertical, horizontal)
- Activation bar at top–rather than bottom–aligned to navigable characteristic.
- Activation bar custom colour support.
- 'Always on' as optional (ie., Bar doesn't have to stay visible/no toggle state).
- 'Level' number to drive label inset amount.
- 'Parent item' as bool to determine presence of arrow (to right in vertical nav, at bottom in horizontal nav).
- Pattern examples of stacked nav items, horizontal and vertical.
- Pattern example of nav item 'nesting'.
- Pattern example of nav item with nested cards.
- Pattern example of nav item with nested mega menu.

**Comment by bhoppers2008** [comment]
Now that we've added Nav Item to the backlog perhaps we can get together @shey-v @JoanaMMoreira to discuss feature requirements to support Vertical and Horizontal Navigation patterns. @joshwooding and I have already started knocking a few requirements together, including:
- Orientation support (vertical, horizontal)
- Activation bar at top–rather than bottom–aligned to navigable characteristic.
- Activation bar custom colour support.
- 'Always on' as optional (ie., Bar doesn't have to stay visible/no toggle state).
- 'Level' number to drive label inset amount.
- 'Parent item' as bool to determine presence of arrow (to right in vertical nav, at bottom in horizontal nav).
- Pattern examples of stacked nav items, horizontal and vertical.
- Pattern example of nav item 'nesting'.
- Pattern example of nav item with nested cards (MCP).
- Pattern example of nav item with nested mega menu (MCP).

**Comment by james-nash** [comment]
For the Salt site, we also need this to collapse into a hamburger menu on narrow viewport sizes.

Also, from an a11y and usability point of view, can we please make sure that nav items use proper `<a>` elements to render links (as opposed to `<div>`s with click handlers or something like that). That way assistive tech can correctly identify and list them as links, search engines can follow them and things like CMD+clicking to open in new tabs and link-specific context menu features (e.g. copy link, open in new window, etc.) will work.

Seems obvious, but in UITK our vertical nav component does _not_ do this, so I'm calling it out here as an explicit requirement. :-)

**Comment by joshwooding** [comment]
> For the Salt site, we also need this to collapse into a hamburger menu on narrow viewport sizes.
> 

This wouldn't be a feature of vertical navigation. It would be functionality on the site in how it builds a drawer

**Comment by james-nash** [comment]
Fair enough. As long as the site's got _a_ way of achieving it, I'm happy ^_^

(From a broader patter documentation perspective though, this might be something to mention though (unless responsive nav is a separate pattern in its own right?). I imagine our site won't be the only one needing to do something like that)

**Comment by parsakhan21** [comment]
what are next steps on this work? @bhoppers2008 @joshwooding

**Comment by shey-v** [comment]
Scope and features have been defined and socialised, with Jo, Ben, Josh and team. Next steps are visual design exploration (currently WIP) and defining Nav item characteristics spec.

**Comment by bhoppers2008** [comment]
Updated... reviewed the scope and added some feedback for @shey-v

**Comment by parsakhan21** [comment]
June 9 - updating feedback ; catch up with josh to review comments/feedback

**Comment by shey-v** [comment]
June 16 - shared work in progress and research collected with team in design review yesterday, have some feedback to address. Continuing to explore visual treatments, characteristics + refine the functional requirements.

**Comment by shey-v** [comment]
June 30 - Visual explorations complete, need to make decisions on which one to go forward with and align to characteristics. Meeting to review scheduled next week Monday with Ben, Josh, Paul, Darrin. Currently working on mocking up the visual aspects of functional requirements following our densities, size and spacing tokens.

**Comment by shey-v** [comment]
June 5 - Design review with Ben, Darrin, Josh and Jo took place and addressing feedback on visual designs and user reqs. WIP.

**Comment by shey-v** [comment]
June 6 - Shey & Jo to refine scope for v1 and another design review is scheduled for Monday with Ben, Jo and Josh.

**Comment by dplsek** [comment]
07/07 - Jo catching up with Shey today, design review scheduled for Monday

**Comment by shey-v** [comment]
14 Jul - Final design specs being created. Design review with Darrin and Ben scheduled for Monday next week.

**Comment by shey-v** [comment]
July 27th - Shey building Figma components after sync up with design leads and agreement on final decisions for nav item v1.

**Comment by JoanaMMoreira** [comment]
July 28 - Caught up with Shey about design updates. Component build and docs are in progress

**Comment by AnniaT** [comment]
On track for delivery next Friday. 

@shey-v finalising all the Figma artefacts. @JoanaMMoreira finalising docs.

**Comment by AnniaT** [comment]
ADA Spec currently being done. An additional example to be added. Reviews to be done

All on track for delivery on Friday

**Comment by parsakhan21** [comment]
August 2 - reviewing build ; finalize mega menu option - catch up with darrin to reviews; shey working on docs; all design assets on track for friday release ; Jo - dev: in progress (PR is up for review), adding one example for mega menu

**Comment by shey-v** [comment]
Aug 4 - Following discussions with Darrin and Ben, the mega menu options and anything 'pattern' related and not component related has been parked for now, so that we have time to define and implement it right, rather than assuming or rushing. 

Has been removed from build examples and the figma docs. Shey / Jo now finalising site docs and Jo updating examples as per design QA. Still on more round of QA to go - Shey currently working on this today. 

Figma component, completed and reviewed by Darrin. Figma guide is ready to go and ADA spec is still to be reviewed. Shey and Jo have a meeting with Stella to go through this today. Also being peer reviewed by Josh. 

As there are still a few bits left, this has now moved to red and will need to be released next week. 

fyi @parsakhan21

**Comment by JoanaMMoreira** [comment]
Aug 4 - I'm refactoring the component as per our chat yesterday so it either renders a link or a button depending on the use case. Will also update the examples after this

**Comment by parsakhan21** [comment]
August 7 - trying to wrap up asap; reviewed with darrin/josh/zhihao re: ada issues - jo making updates to build; site docs being written by Shey (nam item name - sent to content for suggestions

**Comment by shey-v** [comment]
August 8 - waiting on final design review by Ben. Need urgent content review on the site docs - have pinged @mikearildbrown and @sion-docs to help with this.

**Comment by AnniaT** [comment]
Site docs reviewed by @sion-docs

**Comment by AnniaT** [comment]
dev ready to release today
figma & docs to be done by mid next week

**Comment by AnniaT** [comment]
Will be ready for review for @bhoppers2008 on 15/08
Final review to be done by content
@joshwooding to review ADA

**Comment by sion-docs** [comment]
Docs reviewed and updated.

**Comment by shey-v** [comment]
Figma docs (component, guide) still in progress (addressing feedback) and will be completed by end of the week.

**Comment by shey-v** [comment]
**Pending review** @bhoppers2008 

All design (Figma Component, Specs, Guides) completed, ready to release and pending review. 
Content / Site Docs, done and published 
Build (dev) complete and merged.

**Comment by shey-v** [comment]
Currently in review, @bhoppers2008 to merge and release this item on Monday with some others. Prefer to do it all in one go.
